modules/core/src/main/java/org/apache/ignite/internal/processors/cache/query/GridCacheQueryManager.java

```diff
@@ -1411,17 +1411,19 @@ else if (rmvIter)
     }
 
     /**
-     * Process local scan query.
+     * Process local query with results returned in an iterator.
      *
      * @param qry Query.
      * @param updateStatistics Update statistics flag.
      */
     @SuppressWarnings({"unchecked"})
-    protected GridCloseableIterator scanQueryLocal(final GridCacheQueryAdapter qry,
+    protected GridCloseableIterator queryLocal(final GridCacheQueryAdapter qry,
```

Comment:
Let's create 2 methods - `scanQueryLocal` and `indexQueryLocal`. They should reuse some common code.

Reply:
Done.